lib/view/widgets/qr_code_scanner_bottom_sheet.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:superformula_test/core/resources/extensions.dart';
+import 'package:superformula_test/core/theme/app_colors.dart';
+import 'package:superformula_test/view/blocs/qr_code_validation_bloc/qr_code_validation_bloc.dart';
+import 'package:superformula_test/view/widgets/app_loading_widget.dart';
+
+class QRCodeScannerBottomSheet extends StatelessWidget {
+  final String data;
+
+  const QRCodeScannerBottomSheet({super.key, required this.data});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<QRCodeValidationCubit, QRCodeValidationState>(
+      builder: (context, state) {
+        return SizedBox(
+          width: context.screenWidth,
+          height: context.screenHeight * 0.5,
+          child: Padding(
```

Comment:
**suggestion**: since `SizedBox` and `Padding` do **not** depend on the bloc state, move them outside 🙂 Here you are rebuilding them unnecessarily on each state change